### PR TITLE
[core] Deleted method AtomicAssign

### PR DESCRIPTION
**Pull request by maceligueta**
**Description**
It was giving a compilation error in Linux

### CODE REVIEW

**Comment by maceligueta** [comment]
@AlejandroCornejo and @RiccardoRossi detected this error. Weird because it passed all the compilations in the CI.

**Comment by RiccardoRossi** [comment]
does it compile with "atomic write" in win?

**Comment by maceligueta** [comment]
> does it compile with "atomic write" in win?

no, the keyword 'write' does not exist in OpenMP 2.0
https://www.openmp.org/wp-content/uploads/cspec20.pdf

File: kratos/utilities/atomic_utilities.h [hunk 1]
```diff
@@ -65,39 +65,6 @@ inline void AtomicSub(TDataType& target, const TDataType& value ) {
     target -= value;
 }
 
-/** @param target vector variable being atomically updated by doing target -= value
- * @param value vector value being subtracted
- * Note that the update is not really atomic, but rather is done component by component
- */
-template<class TVectorType1, class TVectorType2>
-inline void AtomicSub(TVectorType1& target, const TVectorType2& value ) {
```
Comment (philbucher) [inline kratos/utilities/atomic_utilities.h]: is deleting this method intentional?

**Review by philbucher — COMMENTED** [review]
hm was worth a shot...

**Review by GuillermoCasas — COMMENTED** [review]
Yes, I got a compilation error too. This fixed it.

File: kratos/utilities/atomic_utilities.h (hunk 1 above)
Comment (RiccardoRossi) [reply kratos/utilities/atomic_utilities.h]: atomic sub should remain...what is the problem with this?

Comment (maceligueta) [reply kratos/utilities/atomic_utilities.h]: I'm sorry, I did it too fast and deleted too much. Now it's restored.

**Review by philbucher — APPROVED** [review]
thanks and sorry for the inconvenience everyone
I should have compiled it again locally...
not sure why the CI worked...

**Comment by philbucher** [comment]
merging so that the core compiles again
thx for fixing @maceligueta

**Pull request merged**